<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1040</width>
    <height>860</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1040</width>
    <height>860</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1040</width>
    <height>860</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>10</y>
      <width>241</width>
      <height>128</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>Edges settings</string>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_21">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_19">
       <item>
        <widget class="QLabel" name="label_line">
         <property name="text">
          <string>Type:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_edges_type">
         <item>
          <property name="text">
           <string>Solid</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Dashed</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_20">
       <item>
        <widget class="QLabel" name="label_line_color_2">
         <property name="text">
          <string>Color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_edges_color">
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Black</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Red</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Green</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Blue</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Yellow</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_21">
       <item>
        <widget class="QLabel" name="label_line_width">
         <property name="text">
          <string>Thickness:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_edges_thickness">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>790</x>
      <y>10</y>
      <width>241</width>
      <height>128</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_8">
     <item>
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>Vertices settings</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_22">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_22">
       <item>
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>Type:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_vertices_type">
         <item>
          <property name="text">
           <string>None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Circle</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Square</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_23">
       <item>
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_vertices_color">
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Black</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Red</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Green</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Blue</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Yellow</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_24">
       <item>
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Thickness:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_vertices_thickness">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>15</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>241</width>
      <height>94</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <widget class="QLabel" name="label_23">
       <property name="text">
        <string>Projection settings</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_20">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Type:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_projection_type">
         <item>
          <property name="text">
           <string>Central</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Parallel</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_10">
       <item>
        <widget class="QLabel" name="label_background_color">
         <property name="text">
          <string>Background color:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboBox_background_color">
         <item>
          <property name="text">
           <string>White</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Black</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Red</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Green</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Blue</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Yellow</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>241</width>
      <height>176</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>About file</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_26">
       <item>
        <widget class="QPushButton" name="pushButton_open_file">
         <property name="text">
          <string>Open file</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_27">
       <item>
        <widget class="QLabel" name="label_file_name">
         <property name="text">
          <string>File name:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="file_name">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_28">
       <item>
        <widget class="QLabel" name="label_count_vertexes">
         <property name="text">
          <string>Number of vertexes:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="count_vertexes">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_29">
       <item>
        <widget class="QLabel" name="label_count_facets">
         <property name="text">
          <string>Number of edges:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="count_facets">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line_13">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_set_default_settings">
       <property name="text">
        <string>Reset settings</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="OpenGL" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>200</y>
      <width>1021</width>
      <height>651</height>
     </rect>
    </property>
    <widget class="QPushButton" name="pushButton_move_x_plus">
     <property name="geometry">
      <rect>
       <x>890</x>
       <y>610</y>
       <width>40</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⇨</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_move_x_minus">
     <property name="geometry">
      <rect>
       <x>840</x>
       <y>610</y>
       <width>40</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⇦</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_rotate_x_minus">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>610</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⤺</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_rotate_x_plus">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>610</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⤻</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_rotate_y_plus">
     <property name="geometry">
      <rect>
       <x>980</x>
       <y>610</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>↺</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_rotate_y_minus">
     <property name="geometry">
      <rect>
       <x>950</x>
       <y>610</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>↻</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_scale_plus">
     <property name="geometry">
      <rect>
       <x>950</x>
       <y>580</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>+</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_move_y_minus">
     <property name="geometry">
      <rect>
       <x>980</x>
       <y>520</y>
       <width>25</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⇩</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_scale_minus">
     <property name="geometry">
      <rect>
       <x>980</x>
       <y>580</y>
       <width>25</width>
       <height>25</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>-</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_move_y_plus">
     <property name="geometry">
      <rect>
       <x>980</x>
       <y>470</y>
       <width>25</width>
       <height>40</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="text">
      <string>⇧</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>OpenGL</class>
   <extends>QWidget</extends>
   <header location="global">opengl.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
